src/shared-api-objects/current-user.ts

```diff
@@ -0,0 +1,23 @@
+import { BehaviorSubject } from "rxjs";
+import { openmrsObservableFetch, FetchResponse } from "../openmrs-fetch";
+import { mergeAll, filter, map } from "rxjs/operators";
+
+const userSubject = new BehaviorSubject(
+  openmrsObservableFetch("/ws/rest/v1/session")
+);
+
+export function getCurrentUser(opts: CurrentUserOptions = {}) {
+  return userSubject.asObservable().pipe(
+    mergeAll(),
+    map((r: FetchResponse) => r.data),
+    filter(user => opts.includeUnauthenticated || user.authenticated)
```

Comment:
Right now this returns the session object which includes the user but also includes properties like session id. Should this just return the user object? In other words, return r.data.user? If so, note that the following filter would have to go before hand.

Reply:
@joeldenning could you explain what this function does? I dont understand why you need the `filter`.

Reply:
>  In other words, return r.data.user

👍 updated to do this

> could you explain what this function does? I dont understand why you need the filter

The `/ws/rest/v1/session` endpoint returns a 200/304 no matter what, even if you're not logged in. It would be pretty unintuitive for the `getCurrentUser` observable to produce a value that isn't even a user. What `filter` does is similar to Array.prototype.filter does -- it skips any values where the user is not authenticated.